src/scripts/main.js

```diff
@@ -1,3 +1,231 @@
+    gameField[randomRow][randomCell] = smallDigit;
+    updeteInfo();
+  } else {
+    udateGameField();
+    checkforfGameOver();
+  }
+
+  const arrOfNumbers = [];
+
+  for (const arr of gameField) {
+    for (const num of arr) {
+      arrOfNumbers.push(num)
+    }
+  }
+
+  let maxCellNum = Math.max(...arrOfNumbers);
+
+  if (maxCellNum === 2048) {
+    messageWin.classList.remove('hidden');
+  }
+
+};
+
+function updeteInfo() {
+  for (let row = 0; row < rows; row++) {
+    for (let cell = 0; cell < columns; cell++) {
+      if (gameField[row][cell] === 0) {
+        fieldRow[row].children[cell].textContent = '';
+        fieldRow[row].children[cell].className = 'field-cell';
+      } else {
+        fieldRow[row].children[cell].textContent = gameField[row][cell];
+        fieldRow[row].children[cell].className = 'field-cell';
+
+        fieldRow[row].children[cell]
+          .classList.add(`field-cell--${gameField[row][cell]}`);
+      }
+    }
+  }
+}
+
+function checkforfGameOver() {
+  let emptyCells = 0;
+
+  for (let row = 0; row < rows; row++) {
+    for (let cell = 0; cell < columns; cell++) {
+      if (gameField[row][cell] === 0) {
+        emptyCells++;
+      }
+    }
+  }
+
+  if (emptyCells === 0 && !possiblesMovements()) {
+    messageGameOver.classList.remove('hidden');
+  }
+}
+
+function possiblesMovements() {
+  for (let row = 0; row < rows - 1; row++) {
+    for (let cell = 0; cell < columns; cell++) {
+      if (gameField[row][cell] === gameField[row + 1][cell]) {
+        return true;
+      }
+    }
+  }
+
+  for (let row = 0; row < rows; row++) {
+    for (let cell = 0; cell < columns - 1; cell++) {
+      if (gameField[row][cell] === gameField[row][cell + 1]) {
+        return true;
+      }
+    }
+  }
+}
+
+button.addEventListener('click', () => {
+  button.classList.remove('start');
+  button.classList.add('restart');
+
+  messageStart.classList.add('hidden');
+  messageGameOver.classList.add('hidden');
+  messageWin.classList.add('hidden');
+
+  if (button.classList.contains('restart')) {
+    button.textContent = 'Restart';
+    restart();
+  }
+
+  udateGameField();
+  udateGameField();
+});
+
+function restart() {
+  for (let row = 0; row < rows; row++) {
+    for (let cell = 0; cell < columns; cell++) {
+      gameField[row][cell] = 0;
+    }
+  }
+}
+
+document.addEventListener('keydown', function(ev) {
+  const key = ev.key;
+
+  if (button.classList.contains('restart')) {
+    switch (key) {
+      case 'ArrowLeft':
+        moveCells('left');
+        udateGameField();
+        break;
+      case 'ArrowRight':
+        moveCells('right');
+        udateGameField();
+        break;
+      case 'ArrowUp':
+        moveCells('up');
+        udateGameField();
+        break;
+      case 'ArrowDown':
+        moveCells('down');
+        udateGameField();
+        break;
+    }
+  }
+});
+
+function moveCells(direction) {
+  const gameFieldReversed = [
+    [],
+    [],
+    [],
+    [],
+  ];
```

Comment:
looks strange. Try smth like this
```suggestion
  const gameFieldReversed = new Array(4).fill([])
```

Reply:
it works weird. I understood the idea, but I think that for this situation it's not okay.